src/stores/FlairStore.js

```diff
@@ -154,16 +154,26 @@ class FlairStore extends EventEmitter {
             return this._groupProfiles[groupId];
         }
 
-        // No request yet, start one
-        if (!this._groupProfilesPromise[groupId]) {
-            this._groupProfilesPromise[groupId] = matrixClient.getGroupProfile(groupId);
+        // A request is ongoing, wait for it to complete and return the group profile.
+        if (this._groupProfilesPromise[groupId]) {
+            try {
+                await this._groupProfilesPromise[groupId];
+            } catch (e) {
+                // Don't log the error; this is done below
+                return undefined;
+            }
+            return this._groupProfiles[groupId];
         }
 
+        // No request yet, start one
+        console.log('FlairStore: Request group profile of ' + groupId);
+        this._groupProfilesPromise[groupId] = matrixClient.getGroupProfile(groupId).delay(5000);
```

Comment:
why the delay?

Reply:
Oops, that shouldn't be there, I was attempting to exacerbate the race.